e2e/testcases/git_sync_test.go

```diff
@@ -35,7 +35,7 @@ func TestMultipleRemoteBranchesOutOfSync(t *testing.T) {
 	}
 
 	nt.T.Log("Create an extra remote tracking branch")
-	nt.Must(nt.RootRepos[configsync.RootSyncName].Push("HEAD:refs/remotes/upstream/main"))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Push("HEAD:upstream/main"))
```

Comment:
I believe this may not be functionally the same. IIUC In this context `upstream/main` will just be the name of a head/branch on the remote, whereas before we were pushing a remote ref. If the fix was reverted, would this test still reproduce the bug?